presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryColumnWriter.java

```diff
@@ -71,7 +75,7 @@
     private final List<DictionaryRowGroup> rowGroups = new ArrayList<>();
     private long columnStatisticsRetainedSizeInBytes;
 
-    private IntBigArray rowGroupIndexes;
+    private int[] rowGroupIndexes;
```

Comment:
Is this safe? We will face gigantic int[] in memory. We used to have the problem for Presto on Spark. cc: @arhimondr

Reply:
rowGroupIndexes are typically 10K and I haven't seen it configured ever. IntBigArray has 1024 per segment and 1024 segments. Only 10 segments are used today and rest 1014 segments point to null. So per 10K rows, we are wasting 4KB with using IntBigArray there.